code/game/objects/items/stacks/predator.dm

```diff
@@ -15,41 +15,45 @@
 	max_amount = 8
 
 /obj/item/stack/yautja_rope/attack(mob/living/mob_victim, mob/living/carbon/human/user)
-	. = ..()
-	if(!.)
-		return
+	if(mob_victim.stat != DEAD)
+		return ..()
+
+	if(mob_victim.mob_size != MOB_SIZE_HUMAN)
+		to_chat(user, SPAN_WARNING("[mob_victim] has the wrong body plan to hang up."))
+		return TRUE
 
 	if(!HAS_TRAIT(user, TRAIT_SUPER_STRONG))
 		to_chat(user, SPAN_WARNING("You're not strong enough to lift [mob_victim] up with a rope. Also, that's kind of fucked up."))
-		return
-
-	if(mob_victim.mob_size != MOB_SIZE_HUMAN)
-		return
+		return TRUE
 
 	var/mob/living/carbon/human/victim = mob_victim
-	if(victim.stat != DEAD)
-		return
 
 	if(!do_after(user, 1 SECONDS, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE, victim))
-		return
+		return TRUE
+
+	user.visible_message(SPAN_NOTICE("[user] starts to secure \his rope to the ceiling..."),
+		SPAN_NOTICE("You start securing the rope to the ceiling..."))
 
-	to_chat(user, SPAN_NOTICE("You start securing the rope to the ceiling..."))
 	if(do_after(user, 4 SECONDS, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE, victim))
 		var/turf/rturf = get_turf(victim)
 		var/area/rarea = get_area(victim)
 		if(rturf.density)
 			to_chat(user, SPAN_WARNING("They're in a wall!"))
-			return
+			return TRUE
 		if(rarea.ceiling == CEILING_NONE)
-			to_chat(user, SPAN_WARNING("There's no ceiling!"))
-			return
-		to_chat(user, SPAN_NOTICE("You secure the rope."))
-		to_chat(user, SPAN_NOTICE("You start hanging [victim] by the rope..."))
-		if(do_after(user, 4 SECONDS, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE, victim))
-			to_chat(user, SPAN_NOTICE("You finish the hanging of [victim]."))
-			user.stop_pulling()
-			victim.get_hung()
-			use(1)
+			to_chat(user, SPAN_WARNING("There's no ceiling to hang them from!"))
+			return TRUE
+		user.visible_message(SPAN_NOTICE("[user] secures the rope."),
+			SPAN_NOTICE("You secure the rope."))
+		if(do_after(user, 1 SECONDS, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE, victim))
+			user.visible_message(SPAN_WARNING("[user] begins hanging [victim] up by the rope..."),
+				SPAN_NOTICE("You start hanging [victim] up by the rope..."))
+			if(do_after(user, 3 SECONDS, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE, victim))
+				user.visible_message(SPAN_WARNING("[user] hangs [victim] from the ceiling!"), SPAN_NOTICE("You finish hanging [victim]."))
+				user.stop_pulling()
+				victim.get_hung()
+				use(1)
```

Comment:
use early returns here. You can just return without doing anything else if the do_after fails

Reply:
i am not sure about where to put it in the indentation so .... i am putting it there

Reply:
Look at how other do_afters with early returns work, like the new on line 32.

Reply:
early returning is more about removing the extra indents
and replacing your if statements with the negated version of itself, where you then return early instead of indenting the entire block
It's the difference between
```dm
if(do_after(user, 1 SECONDS, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE, victim))
    ...
```
and
```dm
if(!do_after(user, 1 SECONDS, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE, victim))
    return
...```

Reply:
that's smart

Reply:
i flatten the indentation...